EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -374,6 +380,43 @@ int RawDataUnpacker::ProcessVFATDataParallel(const uint16_t *buf, unsigned int O
     presenceFlags |= 0x8;
     fd[0] = buf[dataOffset + 8];
   }
+  
+  // get channel data for diamond compact mode
+  if (hFlag == vmDiamondCompact)
+  {
+    for (unsigned int i = 1; (buf[i+1] & 0xFFF0)!= 0xF000; i++) {
```

Comment:
Loop protected in https://github.com/cms-sw/cmssw/pull/19754/commits/a2ba7c69467364715812ad7c65ebe2e284485dc9